feat: added auto curVersion with device info

ShivamJoker opened the pull request:
This PR makes the curVersion parameter optional as it takes the app version using rn-device-info.

Also mentioned in - #15

ShivamJoker commented:
@SudoPlz can you please review the PR and let me know if there is any change required.

SudoPlz commented on src/InAppUpdates.android.ts:
```diff
@@ -90,11 +91,9 @@ export default class InAppUpdates extends InAppUpdatesBase {
-        'checkNeedsUpdate'
-      );
+    let appVersion: string = getVersion();
+    if (curVersion) {
```
Instead of always running `getVersion` (even if `curVersion` exists) could we only run it if it doesn't?

ShivamJoker replied:
sure why not, let me try

SudoPlz commented on src/InAppUpdates.ios.ts:
```diff
@@ -18,11 +19,10 @@ export default class InAppUpdates extends InAppUpdatesBase {
+
+    let appVersion: string = getVersion();
+    if (curVersion) {
+      appVersion = curVersion;
```
Same as above

SudoPlz reviewed (changes requested):
This looks good, have you tested it locally? Is it working fine?

ShivamJoker commented:
How should I test it ? I don't see any test files

SudoPlz commented:
I mean locally, using the example project.
Also what's with the yarn file changes, was that intended?

Looking good so far.

ShivamJoker commented:
<img width="472" alt="image" src="https://user-images.githubusercontent.com/23727670/117028350-43507280-ad1b-11eb-95c1-dba2b3242853.png">

I am getting this on android 
And ios build is failing for some reason (maybe due to latest xcode update)

ShivamJoker commented:
Okay after fixing pods again it seems to work tested it with and without passing version.
Hope now you can merge it

<img width="520" alt="image" src="https://user-images.githubusercontent.com/23727670/117036729-60893f00-ad23-11eb-84b3-78226fcf0ffa.png">

SudoPlz commented:
Cool, thanks for the changes, looking good so far.
Question, what's with the yarn file changes, was that intended?

ShivamJoker commented:
I removed yarn files because I added it accidentally, seems you are using
npm instead of yarn.

On Wed, May 5, 2021, 5:10 PM Ioannis Kokkinidis ***@***.***>
wrote:

> Cool, thanks for the changes, looking good so far.
> Question, what's with the yarn file changes, was that intended?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SudoPlz/sp-react-native-in-app-updates/pull/20#issuecomment-832620589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFVA4NWJL62MJWCAQHBQ37TTMEVCNANCNFSM44CUFJNQ>
> .
>

SudoPlz commented:
I still see yarn files in your pr

ShivamJoker commented:
is it fine now ?

SudoPlz commented:
There's still a `.yarnrc` file, could you please delete that too?

Thanks

ShivamJoker commented:
@SudoPlz I have deleted it, it's actually in your repo.

SudoPlz commented:
@ShivamJoker apologies I totally missed that.

Merging now.

SudoPlz commented:
Released under `1.1.0`

ShivamJoker commented:
> Released under `1.1.0`

Wouldn't it be better to use github actions and connect release directly to npm ?

SudoPlz commented:
Do you feel like creating a pr for that too?

ShivamJoker commented:
@SudoPlz we are still needing to pass empty object, can you fix it ?

SudoPlz commented:
@ShivamJoker that was just a typings issue. Fixed it on `1.1.1`
